Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -142,8 +145,10 @@ private extension TabScrollingController {
 
             lastContentOffset = translation.y
             if checkRubberbandingForDelta(delta) && checkScrollHeightIsLargeEnoughForScrolling() {
-                if toolbarState != .Collapsed || contentOffset.y <= 0 {
-                    scrollWithDelta(delta)
+                if toolbarState != .Collapsed || contentOffset.y <= topScrollHeight {
+                    if !surpressToolbarHiding {
```

Comment:
Nit: You can probably combine this if clause with the above:

`(toolbarState != .Collapsed || contentOffset.y <= topScrollHeight) && !surpressToolbarHiding`